processors/data_contributions/processor.py

```diff
@@ -0,0 +1,125 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import requests
+import logging
+from .. import base
+logger = logging.getLogger(__name__)
+
+
+def process(conf, conn):
+    def convert_ids(contrib):
+        converted_ids = {key: val.hex for key, val in contrib.items()
+                        if val.__class__.__name__ == 'UUID'}
+        contrib.update(converted_ids)
+        return contrib
+
+    source_id = _create_source(conn)
+    for_processing = """SELECT * FROM data_contributions
+                        WHERE approved=true
+                        AND trial_id IS NOT NULL
+                        AND data_category_id IS NOT NULL
+                        AND (data_url IS NOT NULL OR url IS NOT NULL);"""
+
+    for data_contrib in conn['explorer'].query(for_processing):
+        data_contrib = convert_ids(data_contrib)
+        trial = conn['database']['trials'].find_one(id=data_contrib['trial_id'])
+        if not trial:
+            continue
+        _process_document(conn, data_contrib, source_id)
+
+    for_removal = """SELECT * FROM data_contributions
+                     WHERE approved=false
+                     AND document_id IS NOT NULL;"""
+    for data_contrib in conn['explorer'].query(for_removal):
+        data_contrib = convert_ids(data_contrib)
+        _remove_document(conn, data_contrib)
+
+
+def _process_document(conn, contrib, source_id):
+    document_id = None
+    data_url = contrib['data_url'] or contrib['url']
```

Comment:
@vitorbaptista I didn't understand if a contribution can have `data_url OR url` or `data_url AND url`. Can you please explain?